hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpcengine/net/AsyncSocketOptions.java

```diff
@@ -70,6 +72,10 @@ public interface AsyncSocketOptions {
      */
     Option<Integer> TCP_KEEPCOUNT = new Option<>("TCP_KEEPCOUNT", Integer.class);
 
+    Option<Object> SSL_ENGINE_FACTORY = new Option<>("SSL_ENGINE_FACTORY", Object.class);
```

Comment:
```suggestion
    Option<SSLEngineFactory> SSL_ENGINE_FACTORY = new Option<>("SSL_ENGINE_FACTORY", SSLEngineFactory.class);
```

Reply:
This won't work, because `SSLEngineFactory` is defined in the `hazelcast` module. This code is part of the `hazelcast-tpc-engine` module, which has no dependency on `hazelcast` module. We can't introduce it, because it would be cyclic and we don't want to introduce it, because we want `hazelcast-tpc-engine` not to be dependent on the `hazelcast` and all the other modules.
The fact that it's defined in the `hazelcast` module is hardly an argument, because I just put it there, so I guess I could put it anywhere, i.e. move it to the `hazelcast-tpc-engine` module. However, as much as it would technically work, I believe that `hazelcast-tpc-engine` is not the proper place for this interface.
I think that the ideal way to address this would be to extract another module, which would be a common dependency of `hazelcast` and `hazelcast-tpc-engine` and would contain things like this, a _shared kernel_ if you will. This however is very well beyond the scope of this PR and this story in general.
Finally, this is a deep internal implementation detail, not really meant to be used other than in the TPC sockets and TPC bootstapping code.

Perhaps @pveentjer has a few things to add here.